### PR TITLE
Fix problem bug in MobileBert with vectorization enable.

### PR DESCRIPTION
**Pull request by ThomasRaoux**

### CODE REVIEW

File: iree/compiler/Conversion/LinalgToSPIRV/KernelDispatchUtils.cpp [hunk 1]
```diff
@@ -351,21 +360,14 @@ LogicalResult getOpLaunchConfig(linalg::GenericOp op,
...
-
-  // Skip vectorization for non-minor identity inputs as it generates
-  // transfer_read ops with permutation maps that we currently cannot lower.
-  // TODO: Remove this restriction once the lowering of the permutation map is
-  // supported in core.
-  for (unsigned i = 0, e = op.getNumInputs(); i < e; i++) {
-    if (!op.getInputIndexingMap(i).isMinorIdentity()) return success();
-  }
+  if (!vectorize || outputShape.getShape().back() % lowerTs != 0)
+    return success();
```
Comment (antiagainst) [inline iree/compiler/Conversion/LinalgToSPIRV/KernelDispatchUtils.cpp:364]: Don't we want to explicitly set `config.vectorize` to false before returning here?

Comment (ThomasRaoux) [reply iree/compiler/Conversion/LinalgToSPIRV/KernelDispatchUtils.cpp:364]: > explicitly

It is false by default. Do you mean doing it for clarity?

Comment (ThomasRaoux) [reply iree/compiler/Conversion/LinalgToSPIRV/KernelDispatchUtils.cpp:365]: Add it.

**Review by antiagainst — APPROVED** [review]

File: iree/compiler/Conversion/LinalgToSPIRV/KernelDispatchUtils.cpp (hunk 1 above)
Comment (antiagainst) [reply iree/compiler/Conversion/LinalgToSPIRV/KernelDispatchUtils.cpp:365]: Thanks! Yup, it also helps when later we switch vectorization to be on by default.

**Comment by hanhanW** [comment]
It looks like this improves the performance of MobileNetV2 on S20...

https://mako.dev/run?run_key=6441876564475904&~cpu=1&~vmla=1&~vlk=1

@antiagainst could you help verify if ToT does have 15ms performance? I'm hitting some weirdness on S20 in the lab, and I don't have one on local side.

**Comment by hanhanW** [comment]
I tested on S10 and I feel that there are issues in benchmarking infrastructure...

**Comment by hanhanW** [comment]
The variance is bug. There are 8 iterations in one benchmark and the range is 64 ms ~ 90 ms. We probably want to ping GPU frequency for lab phones.

I still feel that there is a performance improvement, so it'd be good if you can help check it.

**Comment by antiagainst** [comment]
> @antiagainst could you help verify if ToT does have 15ms performance? 

The ToT is still going through the existing buffer-based path. This PR actually fixed the broken connection to let the existing buffer-based path to be able to access known-good tiling configurations again (for both MobileBERT and MobileNet) so we should see an improvement. But I don't think we can achieve 15ms yet. It should be around 20ms. So looks like we are over optimistic here.

 I think it's likely because we duplicate dispatches without inserting any barriers. That essentially means increasing the number of warps for each dispatch to help filling the GPU. After dividing by the repeat count, we are getting a better number than it should be. There are certain dispatches in both models that are just working on small scale problem size; increasing warp count hides and "improves" the latency for them. Instead of duplicating dispatches, I think we want to submit the command buffer multiple times and average at that level. It will help to keep the original dispatch structure intact.

**Pull request merged**